lib/Compiler.php

```diff
@@ -0,0 +1,195 @@
+<?php
+namespace Goetas\Xsd\XsdToPhp;
+
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Process\Process;
+
+/**
+ * The Compiler class compiles xsd2php into a phar
+ */
+class Compiler
+{
+    private $version;
+    private $versionDate;
+
+    /**
+     * Compiles xsd2php into a single phar file
+     *
+     * @throws \RuntimeException
+     * @param string $pharFile The full path to the file to create
+     */
+    public function compile($pharFile = 'xsd2php.phar')
+    {
+        if (file_exists($pharFile)) {
+            unlink($pharFile);
+        }
+
+        $process = new Process('git log --pretty="%H" -n1 HEAD', __DIR__);
+        if ($process->run() != 0) {
+            throw new \RuntimeException('Can\'t run git log. You must ensure to run compile from xsd2php git repository clone and that git binary is available.');
+        }
+        $this->version = trim($process->getOutput());
+
+        $process = new Process('git log -n1 --pretty=%ci HEAD', __DIR__);
+        if ($process->run() != 0) {
+            throw new \RuntimeException('Can\'t run git log. You must ensure to run compile from xsd2php git repository clone and that git binary is available.');
+        }
+
+        $date = new \DateTime(trim($process->getOutput()));
+        $date->setTimezone(new \DateTimeZone('UTC'));
+        $this->versionDate = $date->format('Y-m-d H:i:s');
```

Comment:
Appears to have been forgotten when he altered https://github.com/composer/composer/blob/master/src/Composer/Compiler.php

Originally used here: https://github.com/composer/composer/blob/master/src/Composer/Compiler.php#L169